demo/lightning/checkpoint/simulated/multiprocessing_train.py

```diff
@@ -0,0 +1,190 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ """
+import argparse
+import os
+import time
+import statistics
+from typing import Optional, Dict
+
+import torch
+import torch.nn as nn
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.distributed.checkpoint as dist_cp
+from lightning.pytorch.strategies import FSDPStrategy
+from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedWriter, GCSDistributedReader
+
+MASTER_ADDR = 'localhost'
+MASTER_PORT = '12355'
+
+
+class BenchmarkStrategy(FSDPStrategy):
+    def __init__(self, project: str, path: str, model: nn.Module, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.writer = GCSDistributedWriter(path, project, None)
+        self.reader = GCSDistributedReader(path, project, None)
+        self.model = model
+
+    def save_checkpoint(self, checkpoint: Dict[str, torch.Tensor], filepath: str, storage_options: Optional[Dict] = None) -> None:
+        dist_cp.save(state_dict=checkpoint, checkpoint_id=filepath,
+                     storage_writer=self.writer)
+
+    def load_checkpoint(self, checkpoint_path: str) -> None:
+        empty_state_dict = {}
+        dist_cp.load(state_dict=empty_state_dict,
+                     checkpoint_id=checkpoint_path, storage_reader=self.reader)
+
+
+class SimpleModel(nn.Module):
+    def __init__(self, size: int, padding_size: int) -> None:
+        super(SimpleModel, self).__init__()
+        self.fc1 = nn.Linear(size, size)
+        self.fc2 = nn.Linear(size, size)
+        self.dummy_tensors = [torch.randn(size, size)
+                              for _ in range(padding_size)]
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.fc2(torch.relu(self.fc1(x)))
+
+
+def parse_args() -> argparse.Namespace:
+    """Parses command line arguments."""
+    parser = argparse.ArgumentParser(
+        description="Benchmark distributed checkpointing.")
+    parser.add_argument("--project", type=str,
+                        required=True, help="GCS project ID.")
+    parser.add_argument("--ckpt-dir-path", type=str, required=True,
+                        help="Path to GCS bucket for checkpoints.")
+    parser.add_argument("--model-size", type=int,
+                        default=100, help="Size of the model.")
+    parser.add_argument("--clear-kernel-cache", action="store_true",
+                        default=False, help="Clear kernel cache.")
+    parser.add_argument("--sample-count", type=int, default=3,
+                        help="Number of samples for benchmarking.")
+    parser.add_argument("--padding-size", type=int, default=1000,
+                        help="Size of dummy tensors for padding, to control the size of the checkpoint.")
```

Comment:
Is there any way to add some info on how this setting affects the size? e.g. "1000 is 10 MB" or something like that?

Reply:
Added. It approximately adds 3 MB per 100 units of padding-size.